### PR TITLE
Change tuple tokens and update datatypes theory ref

### PR DESCRIPTION
**Pull request by ajreynol**
Changes `tuple_` to `tuple.` for consistency.

Also fixes the lexer list for smt2 and updates the datatypes theory reference with missing content.

### CODE REVIEW

File: docs/ext/smtliblexer.py [hunk 1]
```diff
@@ -62,15 +62,16 @@ class SmtLibLexer(RegexLexer):
         '<', '>', '<=', '>=', '!=', '\+', '-', '\*', '/', 'div', 'mod', 'abs',
         'divisible', 'to_real', 'to_int', 'is_int', 'iand', 'int2bv',
         # separation logic
-        'emp', 'pto', 'sep', 'wand', 'nil',
+        'sep\.emp', 'pto', 'sep', 'wand', 'sep\.nil',
```
Comment (aniemetz) [inline docs/ext/smtliblexer.py:65]: Should we also rename `pto` and `wand`?

File: test/regress/cli/regress1/quantifiers/bug822.smt2 [hunk 2]
```diff
@@ -774,8 +774,8 @@ Publications: "A Decision Procedure for (Co)Datatypes" by Andrew Reynolds and Ja
 (declare-fun dtor_unfold_tree$a (B_b_b_b_prod_prod_fun$) B_b_tree_fun$)
 (declare-fun dtor_unfold_tree$b (B_tree_a_b_tree_b_tree_prod_prod_fun$) B_tree_a_tree_fun$)
 (declare-fun dtor_unfold_tree$c (B_tree_b_b_tree_b_tree_prod_prod_fun$) B_tree_b_tree_fun$)
-(declare-fun iso_tuple_update_accessor_eq_assist$ (B_b_fun_b_b_fun_fun$) B_b_fun_b_b_b_fun_b_b_bool_fun_fun_fun_fun_fun$)
-(declare-fun iso_tuple_update_accessor_eq_assist$a (B_tree_b_tree_fun_b_tree_b_tree_fun_fun$) B_tree_b_tree_fun_b_tree_b_tree_b_tree_fun_b_tree_b_tree_bool_fun_fun_fun_fun_fun$)
+(declare-fun iso_tuple.update_accessor_eq_assist$ (B_b_fun_b_b_fun_fun$) B_b_fun_b_b_b_fun_b_b_bool_fun_fun_fun_fun_fun$)
+(declare-fun iso_tuple.update_accessor_eq_assist$a (B_tree_b_tree_fun_b_tree_b_tree_fun_fun$) B_tree_b_tree_fun_b_tree_b_tree_b_tree_fun_b_tree_b_tree_bool_fun_fun_fun_fun_fun$)
```
Comment (aniemetz) [inline test/regress/cli/regress1/quantifiers/bug822.smt2:778]: I think this was unintentional?

File: test/regress/cli/regress1/quantifiers/bug822.smt2 [hunk 3]
```diff
@@ -1114,8 +1114,8 @@ Publications: "A Decision Procedure for (Co)Datatypes" by Andrew Reynolds and Ja
...
 (assert (forall ((?v0 B_b_tree_b_tree_prod_prod$)) (! (= (id_bnf$a ?v0) ?v0) :pattern ((id_bnf$a ?v0)))))
 (assert (forall ((?v0 B_b_tree_a_b_fun_sum_b_tree_a_b_fun_sum_prod_prod$)) (! (= (id_bnf$ ?v0) ?v0) :pattern ((id_bnf$ ?v0)))))
-(assert (forall ((?v0 B$) (?v1 B_b_fun$) (?v2 B$)) (=> (= ?v0 (fun_app$aa ?v1 ?v2)) (fun_app$ew (fun_app$ex (fun_app$ey (fun_app$ez (fun_app$fa (iso_tuple_update_accessor_eq_assist$ id$d) id$) ?v2) ?v1) ?v0) ?v2)) ))
-(assert (forall ((?v0 B_tree$) (?v1 B_tree_b_tree_fun$) (?v2 B_tree$)) (=> (= ?v0 (fun_app$x ?v1 ?v2)) (fun_app$dg (fun_app$fb (fun_app$fc (fun_app$fd (fun_app$fe (iso_tuple_update_accessor_eq_assist$a id$c) id$a) ?v2) ?v1) ?v0) ?v2)) ))
+(assert (forall ((?v0 B$) (?v1 B_b_fun$) (?v2 B$)) (=> (= ?v0 (fun_app$aa ?v1 ?v2)) (fun_app$ew (fun_app$ex (fun_app$ey (fun_app$ez (fun_app$fa (iso_tuple.update_accessor_eq_assist$ id$d) id$) ?v2) ?v1) ?v0) ?v2)) ))
+(assert (forall ((?v0 B_tree$) (?v1 B_tree_b_tree_fun$) (?v2 B_tree$)) (=> (= ?v0 (fun_app$x ?v1 ?v2)) (fun_app$dg (fun_app$fb (fun_app$fc (fun_app$fd (fun_app$fe (iso_tuple.update_accessor_eq_assist$a id$c) id$a) ?v2) ?v1) ?v0) ?v2)) ))
```
Comment (aniemetz) [inline test/regress/cli/regress1/quantifiers/bug822.smt2:1118]: Ditto

File: docs/ext/smtliblexer.py (hunk 1 above)
Comment (ajreynol) [reply docs/ext/smtliblexer.py:65]: `sep` would also be renamed to `sep.star` then.

Comment (ajreynol) [reply docs/ext/smtliblexer.py:65]: I guess we should do this.  This contradicts the syntax from https://sl-comp.github.io/docs/smtlib-sl.pdf although I'm not sure the syntax there should be considered stable.

Comment (ajreynol) [reply docs/ext/smtliblexer.py:65]: If we do this, I'll do this in a separate PR.

Comment (aniemetz) [reply docs/ext/smtliblexer.py:65]: Sounds good! I would be in favor of changing this.

**Review by aniemetz — COMMENTED** [review]
It seems that `tuple.project` is missing in the datatypes theory reference.

File: docs/theories/datatypes.rst [hunk 4]
```diff
@@ -176,6 +176,42 @@ a `cvc5::api::Solver solver` object.
...
++--------------------+----------------------------------------+---------------------------------------------------------------------------------------------------------------------------------+
+| Constructor        | ``(Ci <Term_1>, ..., <Term_n>)``       | ``Sort s = solver.mkDatatypeSort(...);``                                                                                        |
+|                    |                                        |                                                                                                                                 |
+|                    |                                        | ``Datatype dt = s.getDatatype();``                                                                                              |
+|                    |                                        |                                                                                                                                 |
+|                    |                                        | ``Term ci = dt[i].getConstructor();``                                                                                           |
```
Comment (aniemetz) [inline docs/theories/datatypes.rst:187]: I think this should be `getConstructorTerm`.

File: docs/theories/datatypes.rst [hunk 5]
```diff
@@ -176,6 +176,42 @@ a `cvc5::api::Solver solver` object.
...
++--------------------+----------------------------------------+---------------------------------------------------------------------------------------------------------------------------------+
+| Selector           | ``(Sij t)``                            | ``Sort s = solver.mkDatatypeSort(...);``                                                                                        |
+|                    |                                        |                                                                                                                                 |
+|                    |                                        | ``Datatype dt = s.getDatatype();``                                                                                              |
+|                    |                                        |                                                                                                                                 |
+|                    |                                        | ``Term sij = dt[i].getSelector(j);``                                                                                            |
```
Comment (aniemetz) [inline docs/theories/datatypes.rst:195]: And this `getSelectorTerm`.

File: docs/theories/datatypes.rst [hunk 6]
```diff
@@ -192,15 +228,31 @@ a `cvc5::api::Solver solver` object.
...
++--------------------+----------------------------------------+---------------------------------------------------------------------------------------------------------------------------------+
+| Tuple Selector     | ``((_ tuple.select i) t)``             | ``Sort s = solver.mkTupleSort(sorts);``                                                                                         |
+|                    |                                        |                                                                                                                                 |
+|                    |                                        | ``Datatype dt = s.getDatatype();``                                                                                              |
+|                    |                                        |                                                                                                                                 |
+|                    |                                        | ``Term sel = dt[0].getSelector(i);``                                                                                            |
```
Comment (aniemetz) [inline docs/theories/datatypes.rst:237]: Ditto

File: docs/theories/datatypes.rst (hunk 4 above)
Comment (ajreynol) [reply docs/theories/datatypes.rst:187]: Yes good catch.

**Review by aniemetz — APPROVED** [review]

**Pull request merged**